src/riak_kv_vnode.erl

```diff
@@ -840,6 +842,24 @@ handle_command({get_index_entries, Opts},
             {reply, ignore, State}
     end;
 
+%% For now, ignore async_put
+handle_command(?KV_W1C_BATCH_PUT_REQ{objs=Objs, type=Type},
+                From, State=#state{mod=Mod, idx=Idx, modstate=ModState}) ->
+    StartTS = os:timestamp(),
+    Context = {w1c_batch_put, From, Type, Objs, StartTS},
+    case Mod:batch_put(Context, Objs, [], ModState) of
+        {ok, UpModState} ->
+            lists:foreach(
+              fun({{Bucket, Key}, EncodedVal}) ->
+                      update_hashtree(Bucket, Key, EncodedVal, State),
```

Comment:
It's not safe to skip on intermediate updates and only call `update_hashtree` once, I assume?